src/codecoverage/backend/codecoverage_backend/services/gcp.py

```diff
@@ -159,7 +161,11 @@ def download_report(self, repository, changeset):
         json_path = os.path.join(self.reports_dir, blob.name.rstrip('.zstd'))
         if os.path.exists(json_path):
             logger.info('Report already available', path=json_path)
-            return json_path
+            if force:
+                os.unlink(json_path)
```

Comment:
Nit: You could make `open_report` automatically remove the file when it is invalid, so you can avoid the `force` parameter here. Or you could do it in `get_coverage`.